pages/_app.tsx

```diff
@@ -1,8 +1,24 @@
-import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import "../styles/globals.css"
+import type { AppProps } from "next/app"
+import { UserContext } from "../context/UserContext"
+import { useMemo, useState } from "react"
+import { ToastContainer } from "react-toastify"
+import { BooksContext } from "../context/BooksContext"
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [user, setUser] = useState(null)
```

Comment:
Try creating an `IUser` or `UserType` and use it as the type.

In my experience, it's better to have these kinds of types to have a single source of truth of what is inside the `user`.

Reply:
fixed through reworking the User Context